File: scratch/check_js.py
import esprima
import re

with open('templates/program_sace/facilitator_dashboard.html', 'r', encoding='utf-8') as f:
    text = f.read()

scripts = re.findall(r'<script>(.*?)</script>', text, re.DOTALL)
for i, script in enumerate(scripts):
    # Strip jinja
    script = re.sub(r'\{\{.*?\}\}', '""', script)
    script = re.sub(r'\{%.*?%\}', '', script)
    try:
        esprima.parseScript(script)
        print(f"Script {i+1} is VALID.")
    except Exception as e:
        print(f"Script {i+1} ERROR: {e}")

File: scratch/check_js.js
const fs = require('fs');

const text = fs.readFileSync('templates/program_sace/facilitator_dashboard.html', 'utf8');

// Extract all scripts
const scriptRegex = /<script>([\s\S]*?)<\/script>/gi;
let match;
let scriptContent = '';
while ((match = scriptRegex.exec(text)) !== null) {
    scriptContent += match[1] + '\n';
}

// Remove Jinja tags
scriptContent = scriptContent.replace(/\{\{.*?\}\}/g, '""');
scriptContent = scriptContent.replace(/\{%.*?%\}/g, '');

try {
    new Function(scriptContent);
    console.log("Syntax is valid!");
} catch (e) {
    console.error("Syntax Error: " + e.message);
}
